airbyte-workers/src/main/java/io/airbyte/workers/temporal/SyncWorkflow.java

```diff
@@ -105,35 +112,40 @@ public StandardSyncOutput run(JobRunConfig jobRunConfig,
                                   StandardSyncInput syncInput)
         throws TemporalJobException {
 
-      return new TemporalAttemptExecution<>(workspaceRoot, jobRunConfig, (jobRoot) -> {
-        final IntegrationLauncher sourceLauncher = new AirbyteIntegrationLauncher(
-            sourceLauncherConfig.getJobId(),
-            Math.toIntExact(sourceLauncherConfig.getAttemptId()),
-            sourceLauncherConfig.getDockerImage(),
-            pbf);
-        final IntegrationLauncher destinationLauncher = new AirbyteIntegrationLauncher(
-            destinationLauncherConfig.getJobId(),
-            Math.toIntExact(destinationLauncherConfig.getAttemptId()),
-            destinationLauncherConfig.getDockerImage(),
-            pbf);
-
-        // reset jobs use an empty source to induce resetting all data in destination.
-        final AirbyteSource airbyteSource =
-            sourceLauncherConfig.getDockerImage().equals(WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB) ? new EmptyAirbyteSource()
-                : new DefaultAirbyteSource(sourceLauncher);
-
-        return new DefaultSyncWorker(
-            jobRunConfig.getJobId(),
-            Math.toIntExact(jobRunConfig.getAttemptId()),
-            airbyteSource,
-            new NamespacingMapper(syncInput.getPrefix()),
-            new DefaultAirbyteDestination(destinationLauncher),
-            new AirbyteMessageTracker(),
-            NormalizationRunnerFactory.create(
+      return new TemporalAttemptExecution<>(
```

Comment:
Same thought here, except the even longer anonymous function was even more intimidating hah.

What about something like this?
```
 public StandardSyncOutput run(JobRunConfig jobRunConfig,
                                  IntegrationLauncherConfig sourceLauncherConfig,
                                  IntegrationLauncherConfig destinationLauncherConfig,
                                  StandardSyncInput syncInput)
        throws TemporalJobException {
      final var sourceLauncher = new AirbyteIntegrationLauncher(sourceLauncherConfig.getJobId(), Math.toIntExact(sourceLauncherConfig.getAttemptId()),
          sourceLauncherConfig.getDockerImage(), pbf);
      final var destinationLauncher = new AirbyteIntegrationLauncher(destinationLauncherConfig.getJobId(),
          Math.toIntExact(destinationLauncherConfig.getAttemptId()), destinationLauncherConfig.getDockerImage(), pbf);
      final var srcdockerImage = sourceLauncherConfig.getDockerImage();
      final var dstdockerImage = destinationLauncherConfig.getDockerImage();

      TemporalAttemptExecution<StandardSyncInput, StandardSyncOutput> attempt = new TemporalAttemptExecution<>(
          workspaceRoot,
          jobRunConfig,
          getSyncWorker(jobRunConfig, sourceLauncher, srcdockerImage, destinationLauncher, dstdockerImage, syncInput),
          () -> syncInput,
          new CancellationHandler.TemporalCancellationHandler());
      return attempt.get();
    }

    private CheckedFunction<Path, Worker<StandardSyncInput, StandardSyncOutput>, Exception> getSyncWorker(
                                                                                                          JobRunConfig jobRunConfig,
                                                                                                          IntegrationLauncher sourceLauncher,
                                                                                                          String sourceDockerImage,
                                                                                                          IntegrationLauncher destinationLauncher,
                                                                                                          String dstLauncherImage,
                                                                                                          StandardSyncInput syncInput) {
      return (unusedJobRoot) -> {
        // reset jobs use an empty source to induce resetting all data in destination.
        final AirbyteSource airbyteSource =
            sourceDockerImage.equals(WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB) ? new EmptyAirbyteSource()
                : new DefaultAirbyteSource(sourceLauncher);

        return new DefaultSyncWorker(
            jobRunConfig.getJobId(),
            Math.toIntExact(jobRunConfig.getAttemptId()),
            airbyteSource,
            new NamespacingMapper(syncInput.getPrefix()),
            new DefaultAirbyteDestination(destinationLauncher),
            new AirbyteMessageTracker(),
            NormalizationRunnerFactory.create(
                dstLauncherImage,
                pbf,
                syncInput.getDestinationConfiguration()));
      };
    }
```